lib/promscrape/discovery/kubernetes/api_watcher.go

```diff
@@ -101,11 +101,19 @@ func (aw *apiWatcher) mustStop() {
 	aw.swosByURLWatcherLock.Unlock()
 }
 
-func (aw *apiWatcher) reloadScrapeWorks(uw *urlWatcher, swosByKey map[string][]interface{}) {
+func (aw *apiWatcher) reloadScrapeWorks(uw *urlWatcher, swosByKey map[string][]interface{}, mustReplace bool) {
 	aw.swosByURLWatcherLock.Lock()
-	aw.swosCount.Add(len(swosByKey) - len(aw.swosByURLWatcher[uw]))
-	aw.swosByURLWatcher[uw] = swosByKey
-	aw.swosByURLWatcherLock.Unlock()
+	defer aw.swosByURLWatcherLock.Unlock()
+	if mustReplace {
+		// fast path
+		aw.swosCount.Add(len(swosByKey) - len(aw.swosByURLWatcher[uw]))
+		aw.swosByURLWatcher[uw] = swosByKey
+		return
+	}
+	// todo label
+	for k := range swosByKey {
+		aw.swosByURLWatcher[uw][k] = swosByKey[k]
+	}
```

Comment:
should you also increment `aw.swosCount` after this? It could be that `swosByKey ` contains new keys?

Reply:
I think, it should be done via separate PR. This metric changes looks incorrect for now.

Reply:
Fixed the incorrect `aw.swosCount` calculations in the ada408499917cf4dec7556d13474c281f52c10d1